### ISSUE TITLE
Fix c warnings

### ISSUE THREAD

**Issue by alerque** [simoncozens/libtexpdf#14]
Currently compiling throws a bunch of c warnings. Some of them may be suppressible, others look like they are legitimate concerns. I'm not enough of a C master to know the differences, but one way or another the build should be setup so it doesn't throw any warnings by default. It's really hard to debug downstream packaging this way.

Note some distro build systems by default enable all warnings as errors and you have to setup overrides to get anything with warnings to even go ahead with the build.

**Comment by alerque** [comment]
A lot of the "blame" for this falls on upstream shoulders. If I understand this correctly fixing these would make it harder to sync upstream changes. Is that correct @simoncozens?

**Comment by simoncozens** [comment]
Right. Let’s do the sync first (things may be fixed), and then see if it’s already fixed. If not, we can raise an issue. (texk is on GitHub these days; it used to be just SVN and a mailing list.)

**Comment by dontlaugh** [comment]
Upstream is on GitHub? Do you know where, specifically?

**Comment by vlasakm** [comment]
@dontlaugh

Upstream project name is `dvipdfmx`. It is nowdays part of TeX Live, as explained on [its CTAN page](https://ctan.org/pkg/dvipdfmx). There is a [mailing list](https://lists.tug.org/dvipdfmx). The sources are in subdirectory of the [SVN TeX Live repository](https://tug.org/svn/texlive/trunk/Build/source/texk/dvipdfm-x/). There is [Norbert Preining's git mirror of the full repository](https://git.texlive.info/texlive/), see information about it on [the main page](https://texlive.info/). The TeX Live repository is huge (it contains many years of history of all TeX packages uploaded to CTAN), if you are only interested in `dvipdfmx` it might be better to use the [`texlive-source`](https://github.com/TeX-Live/texlive-source) repository (as advertised by https://texlive.info/). This one is small enough to be hosted on Github. It is essentially the `Build/source` subdirectory of the SVN repo, hence the name.

AFAIK the mailing list is still the primary means of communication (though there may be some Japanese communication channels I don't know about).

**Comment by aminophen** [comment]
I'm a TeX Live contributor and I have a write access to TeX Live repo. If you can send me a patch to fix warnings, it's welcome.
